### ISSUE TITLE
Move telescope_history to `stdpath('state')`

### ISSUE THREAD

**Issue by yochem** [nvim-telescope/telescope.nvim#2745]
**Is your feature request related to a problem? Please describe.**
`$XDG_STATE_HOME` is adopted by NeoVim. It's intention is the following:

> The $XDG_STATE_HOME contains state data that should persist between (application) restarts, but that is not important or portable enough to the user that it should be stored in $XDG_DATA_HOME. It may contain:
> 
> actions history (logs, history, recently used files, …)
> 
> current state of the application that can be reused on a restart (view, layout, open files, undo history, …)

As stated above, this includes history files.

**Describe the solution you'd like**
Change the default path from `stdpath('data') .. '/telescope_history'` to `stdpath('state') .. '/telescope_history'`.

**Describe alternatives you've considered**
Of course just keeping it like this.

**Additional context**
It is of course not as easy as just substituting `stdpath('data')` with `stdpath('state')`. That would introduce the following problems:

1. Users lose their previous history upon updating
2. `stdpath('data')/telescope_history` would keep existing even though it is not used anymore
3. *IF* users would sync their telescope_history (...why?), they would have to update the location

Possible solution:
Have a function that checks if `stdpath('data')/telescope_history` exists. If so, move file to new location. Most users won't notice it, users that sync their telescope history will notice and have to update (with a simple `git add $XDG_STATE_HOME/nvim/telescope_history`, assuming they use Git).

Let me know if you're alright with this change. I can implement it and create a PR.

**Comment by jamestrew** [comment]
I'm inclined to agree with you but... not sure if most people care 🙃

If we were to do this, I think just going to breaking change route is the way to go. Just rip the bandaid off quickly. It shouldn't affect most people anyways I think.

I'll let others chime in though before we make a decision.

**Comment by yochem** [comment]
> I'm inclined to agree with you but... not sure if most people care 🙃

I agree that most people probably don't care, but when a standard is adopted, it feels like the right thing to 100% comply to it.

> If we were to do this, I think just going to breaking change route is the way to go. Just rip the bandaid off quickly. It shouldn't affect most people anyways I think.

Good point. Let's hear what the others have to say indeed.

**Comment by Conni2461** [comment]
I do not care about this at all. Generally speaking i would say this option can already be configured, so why would we care about creating "noice" for our users (because honestly speaking moving a history file for "no" good reason could be considered noice, as we have announce it as breaking change).

Also i hate the idea of shipping a migration tool (on startup function) that automatically does this for us, it just unnecessary code, that might be buggy, if a user has already configured something else, also regards to windows, etc.

But i am not against doing that if we already do some kinda breaking change regarding history, aka saying history v1 is now, dead, we now offer a new history v2 with more features. no backwards compatibility. you lose it regardless of this change. aka implementing https://github.com/nvim-telescope/telescope.nvim/issues/2290

Does that make sense?
